### PR TITLE
add new react version to all pkgs

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

bumps react and the following other packages: 
  "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^14.4.3",
    "@types/react": "^18.0.21",
    "@types/react-dom": "^18.0.2",
    "react-dom": "^18.2.0",
    **"react-helmet-async": "1.3.0",**
    **"framer-motion": "^7.5.1",**
    **"react-resize-detector": "7.1.2"**

also resolve associated build failures

React 18 upgrade guide for reference: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [web-2055]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.5ea335.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.5ea335.0<br/>@codecademy/gamut-kit@0.6.213-alpha.5ea335.0<br/>@codecademy/gamut-labs@33.6.4-alpha.5ea335.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.5ea335.0<br/>@codecademy/gamut-styles@14.1.2-alpha.5ea335.0<br/>@codecademy/gamut-tests@4.0.15-alpha.5ea335.0<br/>@codecademy/gamut@50.8.2-alpha.5ea335.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.5ea335.0<br/>@codecademy/monaco@0.6.93-alpha.5ea335.0<br/>@codecademy/styleguide@61.7.3-alpha.5ea335.0<br/>@codecademy/variance@0.20.4-alpha.5ea335.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.f3486d.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.f3486d.0<br/>@codecademy/gamut-kit@0.6.213-alpha.f3486d.0<br/>@codecademy/gamut-labs@33.6.4-alpha.f3486d.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.f3486d.0<br/>@codecademy/gamut-styles@14.1.2-alpha.f3486d.0<br/>@codecademy/gamut-tests@4.0.15-alpha.f3486d.0<br/>@codecademy/gamut@50.8.2-alpha.f3486d.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.f3486d.0<br/>@codecademy/monaco@0.6.93-alpha.f3486d.0<br/>@codecademy/styleguide@61.7.3-alpha.f3486d.0<br/>@codecademy/variance@0.20.4-alpha.f3486d.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.75cbe7.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.75cbe7.0<br/>@codecademy/gamut-kit@0.6.213-alpha.75cbe7.0<br/>@codecademy/gamut-labs@33.6.4-alpha.75cbe7.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.75cbe7.0<br/>@codecademy/gamut-styles@14.1.2-alpha.75cbe7.0<br/>@codecademy/gamut-tests@4.0.15-alpha.75cbe7.0<br/>@codecademy/gamut@50.8.2-alpha.75cbe7.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.75cbe7.0<br/>@codecademy/monaco@0.6.93-alpha.75cbe7.0<br/>@codecademy/styleguide@61.7.3-alpha.75cbe7.0<br/>@codecademy/variance@0.20.4-alpha.75cbe7.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.52c766.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.52c766.0<br/>@codecademy/gamut-kit@0.6.214-alpha.52c766.0<br/>@codecademy/gamut-labs@33.6.5-alpha.52c766.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.52c766.0<br/>@codecademy/gamut-styles@14.1.2-alpha.52c766.0<br/>@codecademy/gamut-tests@4.0.15-alpha.52c766.0<br/>@codecademy/gamut@50.9.1-alpha.52c766.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.52c766.0<br/>@codecademy/monaco@0.6.94-alpha.52c766.0<br/>@codecademy/styleguide@61.7.4-alpha.52c766.0<br/>@codecademy/variance@0.20.4-alpha.52c766.0<br/></pre>

File: packages/styleguide/.storybook/main.ts [hunk 1]
```diff
@@ -42,6 +42,12 @@ module.exports = {
       configs.css().module.rules
     );
 
+    config.module.rules.push({
+      test: /\.mjs$/,
+      include: /node_modules/,
+      type: 'javascript/auto',
+    });
+
```
Comment (dreamwasp) [inline packages/styleguide/.storybook/main.ts:50]: https://github.com/storybookjs/storybook/issues/16690#issuecomment-971579785

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.b4ac54.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.b4ac54.0<br/>@codecademy/gamut-kit@0.6.214-alpha.b4ac54.0<br/>@codecademy/gamut-labs@33.6.5-alpha.b4ac54.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.b4ac54.0<br/>@codecademy/gamut-styles@14.1.2-alpha.b4ac54.0<br/>@codecademy/gamut-tests@4.0.15-alpha.b4ac54.0<br/>@codecademy/gamut@50.9.1-alpha.b4ac54.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.b4ac54.0<br/>@codecademy/monaco@0.6.94-alpha.b4ac54.0<br/>@codecademy/styleguide@61.7.4-alpha.b4ac54.0<br/>@codecademy/variance@0.20.4-alpha.b4ac54.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.8.3-alpha.d4e81c.0<br/>@codecademy/gamut-illustrations@0.32.3-alpha.d4e81c.0<br/>@codecademy/gamut-kit@0.6.214-alpha.d4e81c.0<br/>@codecademy/gamut-labs@33.6.5-alpha.d4e81c.0<br/>@codecademy/gamut-patterns@0.8.16-alpha.d4e81c.0<br/>@codecademy/gamut-styles@14.1.2-alpha.d4e81c.0<br/>@codecademy/gamut-tests@4.0.15-alpha.d4e81c.0<br/>@codecademy/gamut@50.9.1-alpha.d4e81c.0<br/>@codecademy/markdown-overrides@0.9.13-alpha.d4e81c.0<br/>@codecademy/monaco@0.6.94-alpha.d4e81c.0<br/>@codecademy/styleguide@61.7.4-alpha.d4e81c.0<br/>@codecademy/variance@0.20.4-alpha.d4e81c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.9.5-alpha.df5381.0<br/>@codecademy/gamut-illustrations@0.32.7-alpha.df5381.0<br/>@codecademy/gamut-kit@0.6.244-alpha.df5381.0<br/>@codecademy/gamut-labs@33.8.13-alpha.df5381.0<br/>@codecademy/gamut-patterns@0.8.20-alpha.df5381.0<br/>@codecademy/gamut-styles@14.2.3-alpha.df5381.0<br/>@codecademy/gamut-tests@4.0.20-alpha.df5381.0<br/>@codecademy/gamut@51.5.1-alpha.df5381.0<br/>@codecademy/markdown-overrides@0.9.17-alpha.df5381.0<br/>@codecademy/monaco@0.6.112-alpha.df5381.0<br/>@codecademy/styleguide@62.5.13-alpha.df5381.0<br/>@codecademy/variance@0.20.5-alpha.df5381.0<br/></pre>

**Comment by jakemhiller** [comment]
Replaced by #2613

**Pull request closed without merging**